### DISCUSSION TITLE
Strange unrelated commit message suggestions

### DISCUSSION THREAD

**Discussion by squaredindex** [community/community#113557] [Copilot Conversations]
### Bug

The GPT source control commit suggestions are usually ok, they are probably only fully applicable to what you've just done 25% of the time, but today I've had some very odd suggestions that are so totally wrong there must be a bug.

I'm working on a PHP project and it's giving me suggestions for Python.
After just updating some markup in one of my PHP views I got this suggestion:
```
Refactor print_hello_world function to greet in main.py
```

This project has zero Python files, it doesn't even reference anything Python related.

**Comment by ayim** [comment]
@squaredindex which IDE are you using?

**Reply by squaredindex** [reply]
VSCode - Insiders

**Reply by ayim** [reply]
Tagging @digitarald and @Akash1134 for VSCode awareness

**Comment by Bulletninja** [comment]
The same is happening to me, it's just random generic commit messages

**Comment by TheArKaID** [comment]
I was here looking for the same problem, and got this one. But for me, this strange appeared at least starting this month, where previously Copilot Generate Commit Message worked very well. 

Using VSCode 1.89.1,

**Comment by digitarald** [comment]
We recently added related commits to the prompt so the LLM follows existing formatting. This seems to cause confusion, for some posting here very frequent. The issue is tracked in our repo: https://github.com/microsoft/vscode-copilot-release/issues/1183

**Reply by squaredindex** [reply]
If I understand you correctly you're saying that previous commits I'd made even in other projects could have been added to the prompt that generated the commit message?

If that is what you mean that's strange to me because I don't recall working on a Python project using it at any point, I'm mainly doing JS, PHP, Go. Also the commit message I got was for a very generic hello world in python which I haven't done.
